vhd/lib/libvhd.c

```diff
@@ -119,6 +120,129 @@ old_clear_bit(volatile char *addr, int nr)
 	((uint32_t *)addr)[nr >> 5] &= ~(1 << (nr & 31));
 }
 
+static int
+makedev_from_file(const char *file, dev_t *dev)
+{
+	*dev = 0;
+
+	FILE *f = fopen(file, "r");
+	if (!f)
+		return -1;
+
+	unsigned int dev_major;
+	unsigned int dev_minor;
+	const int ret = fscanf(f, "%u:%u", &dev_major, &dev_minor);
+	fclose(f);
+
+	if (ret != 2)
+		return -1;
+
+	*dev = makedev(dev_major, dev_minor);
+	return 0;
+}
+
+static int
+is_drbd_device(const struct stat *stats)
+{
+	static const int drbd_major = 147;
```

Comment:
Is this guaranteed to always be 147? There is other code that looks up device major ids from `/proc/devices`

Reply:
Normally yes, like said in the official DRBD documentation: https://www.linbit.com/drbd-user-guide/users-guide-drbd-8-4/

> DRBD device
>
> This is a virtual block device managed by DRBD. It has a device major number of 147, and its minor numbers are numbered from 0 onwards, as is customary. Each DRBD device corresponds to a volume in a resource.

>  Very early DRBD versions hijacked NBD’s device major number 43. This is long obsolete; 147 is the LANANA-registered DRBD device major.

Reply:
Yes, thanks, I found the assigned number. I'm just more used to the code in this area dealing with the blktap devices which use the experimental range (and it looks like they can move around a bit).